src/controllers/ChatsController.ts

```diff
@@ -0,0 +1,85 @@
+import UserController from './UserController';
+
+import ChatsAPI from '@api/ChatsAPI';
+import SocketConnection from '@api/SocketConnection';
+import { getDateChats } from '@helpers/getDateChats';
+import store from '@store';
+import { IChatCreate, IChatItem, IChatUsersRequest, IUser } from '@types';
+
+class ChatsController {
+  private api: ChatsAPI;
+  private socket: SocketConnection | null;
+
+  constructor() {
+    this.api = new ChatsAPI();
+    this.socket = null;
+  }
+
+  async setSocketConnection(userId: string, chatId: string) {
+    const { token } = (await this.api.getToken(chatId)) as unknown as Record<string, unknown>;
+
+    const endpoint = `${userId}/${chatId}/${token}`;
+
+    this.socket = new SocketConnection(endpoint);
+  }
+
+  async getChats() {
+    const chats = await this.api.getChats({ offset: 0, limit: 50 });
+
+    getDateChats(chats as unknown as IChatItem[]);
```

Comment:
Явное приведение типов лучше стараться по возможности не использовать, и этого тут можно избежать, если сможешь сделать так:
```suggestion
    const chats = await this.api.getChats<IChatItem[]>({ offset: 0, limit: 50 });

    getDateChats(chats);
```

В [документации](https://www.typescriptlang.org/docs/handbook/2/generics.html) описано то, что тебе нужно